client/stripe-utils/utils.js

```diff
@@ -842,3 +842,22 @@ export const maybeClearBlikCodeValidation = () => {
 		);
 	}
 };
+
+/**
+ * Gets the base font size for both the regular checkout and the Optimized Checkout (which is 2px larger than the default font size).
+ * So it matches the rest of the checkout form when it is scaled down.
+ *
+ * @param {string} defaultFontSize The default font size of the checkout form, e.g. '16px'.
+ * @return {string} The base font size.
+ */
+export const getFontSizeBase = ( defaultFontSize ) => {
+	if ( getStripeServerData()?.isOCEnabled ) {
+		return (
+			parseFloat( defaultFontSize.match( /(\d+(?:\.\d+)?)/ )[ 0 ] ) +
+			2 +
+			'px'
+		);
```

Comment:
Nit: this is _really_ dense and there is a lot going on here. Might it make sense to split this across multiple lines to make it a bit easier to follow? It may also be worth checking that you actually got a match.
```suggestion
	if ( getStripeServerData()?.isOCEnabled ) {
		// Find digits for font size, ignoring decimals.
		const matches = defaultFontSize.match( /(\d+(?:\.\d+)?)/ );
		if ( matches.length > 0 ) {
			return (
				parseFloat( matches[ 0 ] ) +
				2 +
				'px'
			);
		}
```